static/js/stream_create.js

```diff
@@ -150,18 +150,28 @@ function create_stream() {
 
     let invite_only;
     let history_public_to_subscribers;
+    let is_web_public;
     const privacy_setting = $('#stream_creation_form input[name=privacy]:checked').val();
 
+    // TODO: Infeer the variables so there is no need to define all three.
     if (privacy_setting === 'invite-only') {
+        is_web_public = false;
         invite_only = true;
         history_public_to_subscribers = false;
     } else if (privacy_setting === 'invite-only-public-history') {
+        is_web_public = false;
         invite_only = true;
         history_public_to_subscribers = true;
+    } else if (privacy_setting === 'is-web-public') {
```

Comment:
This whole bit of code is ripe for some refactor. We shouldn't have to specify `is_web_public=false` 3 times.

```js
data.is_web_public = ['is-web-public'].includes(prvacy_setting);
data.history_public_to_subscribers = ['is-web-public', 'public', 'invite-only-public-history'].includes(privacy_setting);
data.invite_only = ['invite-only-public-history', 'invite-only'].includes(privacy_setting);
```

Reply:
This can go into a commit of its own, if you want to make this change now.

Reply:
Yes, I leave that TODO to remind me of working on this, so if a stream is `web_public` we just need to pass the `web_public` arguments because of we now for this case `invite_only = False`, `history_public_to_subscribers = True`. There is no much need to define those on the frontend I guess, just pass the relevant values to the back and let it deal with it.

Reply:
There is certain advantage in computing this here and telling the API what to do than having the API try to figure out these parameters.

For example, there isn't a direct field in the database for `invite-only-public-history`; its a combination of two flags.

Reply:
Yeah, but for that case, I would only set the two values instead of three. But is easier to fix it how you showed :)